janusgraph-lucene/src/main/java/org/janusgraph/diskstorage/lucene/LuceneIndex.java

```diff
@@ -265,8 +263,6 @@ private void mutateStores(Map.Entry<String, Map<String, IndexMutation>> stores,
                 writer.updateDocument(new Term(DOCID, documentId), doc);
             }
             writer.commit();
-        } finally {
-            IOUtils.closeQuietly(reader);
```

Comment:
Do you know why this was `closeQuietly`? As I understand it was used to not throw an exception on failed close which is strange but still. Is it OK that `mutateStores` will throw an exception on failed reader closing?

Reply:
I don't know, but if close() throw any exception then full tx will be rolled back. From one hand this is strange, because any write operation done without exception, but another IndexReader [instantiated](https://github.com/JanusGraph/janusgraph/blob/master/janusgraph-lucene/src/main/java/org/janusgraph/diskstorage/lucene/LuceneIndex.java#L227) from IndexWriter, and any exceptions may mean inconsistent state of IndexWriter too

Reply:
I didn't research much but I see that `reader.close()` may produce an exception in some cases. I am just thinking if there is any logic behind swallowing an exception here. Maybe we should always close the reader quietly here and in `restore` method? I just don't know if it is OK to produce an exception on failed reader closing.

Reply:
Ok, I change try-resource to try-finally-closeQuietly